### ISSUE TITLE
Introduce testing library for component testing with Vitest as test runner

### ISSUE THREAD

**Issue by govind-srinidhi** [Adyen/lume#191]
Vue community recommends testing-library to test the components and Vitest as the test runner.

Lume being a component library, it would be nice to have component tests, so that we can track the interface behavior on different actions.

We will have to first check if testing library can be introduced in our current Lume setup and then it works well on both Vue 2/3 versions.

And once the setup is done, we can start rewriting tests using testing-library
